components/omega/src/ocn/VertCoord.h

```diff
@@ -0,0 +1,213 @@
+#ifndef OMEGA_VERTCOORD_H
+#define OMEGA_VERTCOORD_H
+//===-- base/VertCoord.h - vertical coordinate definitions ------*- C++ -*-===//
+//
+/// \file
+/// \brief Contains the vertical mesh variables and methods for Omega
+///
+/// This header defines the VertCoord class which contains information related
+/// to the vertical coordinate, the vertical extent of the mesh, and the active
+/// vertical layers for each ocean column.
+//
+//===----------------------------------------------------------------------===//
+
+#include "Config.h"
+#include "DataTypes.h"
+#include "Decomp.h"
+#include "Error.h"
+#include "HorzMesh.h"
+#include "Logging.h"
+#include "MachEnv.h"
+#include "OmegaKokkos.h"
+
+#include <memory>
+#include <string>
+
+namespace OMEGA {
+
+class VertCoord {
+
+ private:
+   // Variables from Decomp
+   I4 NCellsOwned;
+   I4 NCellsAll;
+   I4 NCellsSize;
+   I4 NEdgesOwned;
+   I4 NEdgesAll;
+   I4 NEdgesSize;
+   I4 NVerticesOwned;
+   I4 NVerticesAll;
+   I4 NVerticesSize;
+   I4 VertexDegree;
+   Array2DI4 CellsOnEdge;
+   Array2DI4 CellsOnVertex;
+
+   std::string MeshFileName;
+   int MeshFileID;
+
+   static VertCoord *DefaultVertCoord;
+   static std::map<std::string, std::unique_ptr<VertCoord>> AllVertCoords;
+
+   // methods
+
+   /// construct a new vertical coordinate object
+   VertCoord(const std::string &Name,  ///< [in] Name for new VertCoord
+             const Decomp *MeshDecomp, ///< [in] associated Decomp
+             Config *Options           ///< [in] configuration options
+   );
+
+   /// define field metadata
+   void defineFields();
+
+   /// read desired quantities from mesh file
+   void readArrays(const Decomp *Decomp ///< [in] Decomp for mesh
+   );
+
+   // Forbid copy and move construction
+   VertCoord(const VertCoord &) = delete;
+   VertCoord(VertCoord &&)      = delete;
+
+ public:
+   // Vertical dimension
+   I4 NVertLayers;
+   I4 NVertLayersP1;
+
+   // Variables computed
+   Array2DReal PressureInterface;
+   Array2DReal PressureMid;
+   Array2DReal ZInterface;
+   Array2DReal ZMid;
+   Array2DReal GeopotentialMid;
+   Array2DReal LayerThicknessTarget;
+
+   HostArray2DReal PressureInterfaceH;
+   HostArray2DReal PressureMidH;
+   HostArray2DReal ZInterfaceH;
+   HostArray2DReal ZMidH;
+   HostArray2DReal GeopotentialMidH;
+   HostArray2DReal LayerThicknessTargetH;
```

Comment:
If we need these arrays on the host we should probably add functions for syncing them with the device versions.